pkg/machinery/labels/validate.go

```diff
@@ -130,3 +130,28 @@ func ValidateLabelValue(value string) error {
 
 	return nil
 }
+
+// ValidateTaints validates that a set of taints is correctly defined.
+func ValidateTaints(taints map[string]string) error {
+	var multiErr *multierror.Error
+
+	keys := maps.Keys(taints)
+	slices.Sort(keys)
+
+	for _, k := range keys {
+		if err := ValidateQualifiedName(k); err != nil {
+			multiErr = multierror.Append(multiErr, err)
+		}
+
+		val, effect, found := strings.Cut(taints[k], ":")
+		if found && effect != "NoSchedule" && effect != "PreferNoSchedule" && effect != "NoExecute" {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("invalid taint effect: %s", effect))
+		}
+
+		if err := ValidateLabelValue(val); err != nil {
+			multiErr = multierror.Append(multiErr, err)
+		}
+	}
+
+	return multiErr.ErrorOrNil()
+}
```

Comment:
Is this correct way of validating taints?

Reply:
you can check with either k8s docs or their code (we can't use it in machinery, but you can get the idea there)

Reply:
Yup, it uses the same logic https://github.com/kubernetes/kubectl/blob/b35935138f5330d299e89b1278b5738487e4f015/pkg/cmd/taint/utils.go#L39